src/Compilers/CSharp/Portable/Utilities/InterceptableLocation.cs

```diff
@@ -0,0 +1,183 @@
+    /// Opaque data which references a call when used as an argument to 'InterceptsLocationAttribute'.
+    /// The value does not require escaping, i.e. it is valid in a string literal when wrapped in " (double-quote) characters.
+    /// </summary>
+    public abstract string Data { get; }
+
+    /// <summary>
+    /// Gets a human-readable representation of the location, suitable for including in comments in generated code.
+    /// </summary>
+    public abstract string GetDisplayLocation();
+}
+
+#pragma warning disable RSEXPERIMENTAL002 // internal usage of experimental API
+/// <summary>
+/// Version 1 of the InterceptableLocation encoding.
+/// </summary>
+internal sealed class InterceptableLocation1 : InterceptableLocation
+{
+    internal const int ContentHashLength = 16;
+
+    private readonly ImmutableArray<byte> _checksum;
+    private readonly string _path;
+    private readonly int _position;
+    private readonly int _lineNumberOneIndexed;
+    private readonly int _characterNumberOneIndexed;
+    private string? _lazyData;
+
+    internal InterceptableLocation1(ImmutableArray<byte> checksum, string path, int position, int lineNumberOneIndexed, int characterNumberOneIndexed)
+    {
+        Debug.Assert(checksum.Length == ContentHashLength);
+        Debug.Assert(path is not null);
+        Debug.Assert(position >= 0);
+        Debug.Assert(lineNumberOneIndexed > 0);
+        Debug.Assert(characterNumberOneIndexed > 0);
+
+        _checksum = checksum;
+        _path = path;
+        _position = position;
+        _lineNumberOneIndexed = lineNumberOneIndexed;
+        _characterNumberOneIndexed = characterNumberOneIndexed;
+    }
+
+    public override string GetDisplayLocation()
+    {
+        // e.g. `C:\project\src\Program.cs(12,34)`
+        return $"{_path}({_lineNumberOneIndexed},{_characterNumberOneIndexed})";
+    }
+
+    public override string ToString() => GetDisplayLocation();
+
+    public override int Version => 1;
+    public override string Data
+    {
+        get
+        {
+            if (_lazyData is null)
+                _lazyData = makeData();
+
+            return _lazyData;
+
+            string makeData()
+            {
+                var builder = PooledBlobBuilder.GetInstance();
+                builder.WriteBytes(_checksum, start: 0, 16);
+                builder.WriteInt32(_position);
+
+                var displayFileName = Path.GetFileName(_path);
+                builder.WriteUTF8(displayFileName);
+
+                var bytes = builder.ToArray();
+                builder.Free();
+                return Convert.ToBase64String(bytes);
+            }
+        }
+    }
+
+    internal static (ReadOnlyMemory<byte> checksum, int position, string displayFileName)? Decode(string? data, Location diagnosticLocation, BindingDiagnosticBag diagnostics)
+    {
+        if (data is null)
+        {
+            diagnostics.Add(ErrorCode.ERR_InterceptsLocationDataInvalidFormat, diagnosticLocation);
```

Comment:
>diagnostics.Add(ErrorCode.ERR_InterceptsLocationDataInvalidFormat, diagnosticLocation); [](http://example.com/codeflow?start=12&length=87)

It looks like `InterceptableLocation1.Decode()` only has one caller, and that each of the error conditions results in the same diagnostic.

Given that, moving the error reporting to the caller.

Reply:
Tracking in https://github.com/dotnet/roslyn/pull/72814#discussion_r1552604820